src/charts/donut/DonutComponent.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import donut from './donutChart';
+
+
+export default class Donut extends Component {
+
+    static PropTypes = {
+        data: PropTypes.array.isRequired,
+
+        colorSchema: PropTypes.arrayOf(PropTypes.string),
+        externalRadius: PropTypes.number,
+        hasFixedHighlightedSlice: PropTypes.boolean,
+        height: PropTypes.number,
+        highlightSliceById: PropTypes.number,
+        internalRadius: PropTypes.number,
+        isAnimated: PropTypes.boolean,
+        margin: PropTypes.shape({
+            top: PropTypes.number,
+            bottom: PropTypes.number,
+            left: PropTypes.number,
+            right: PropTypes.number,
+        }),
+        orderingFunction: PropTypes.func,
+        width: PropTypes.number,
+
+        customMouseOver: PropTypes.func,
+        customMouseMove: PropTypes.func,
+        customMouseOut: PropTypes.func,
+    }
+
+    static defaultProps = {
+        chart: donut,
+        isAnimated: true,
+    }
+
+    constructor(props) {
+        super(props);
+
+        // We want to make this throw an error if no data is provided
+        if (!props.data) {
+            throw new Error('Data is required!');
+        }
+    }
+
+    componentDidMount() {
+        this.props.chart.create(this._rootNode, this.props.data, this._getChartConfiguration());
+    }
+
+    componentDidUpdate() {
+        this.props.chart.destroy(this._rootNode);
+        this.props.chart.create(this._rootNode, this.props.data, this._getChartConfiguration());
```

Comment:
Did you try to use an 'update' function? Was there any problem with it?

Reply:
I didn't let me check this afternoon at home :)

Reply:
Sorry for the delay! I've created a custom component, with the data in the state, and changing it 2 seconds after `componentDidMount`, and It seems both, the `destroy`/`create` and `update` function works, I used this because it's how it works in `StackedAreaComponent`, should I change it to the `update` function?

Reply:
No problem!
We have some issues with update, we will try to change it soon.
For now this will be OK.